core-test/src/test/scala/org/bitcoins/core/api/CallbackTest.scala

```diff
@@ -16,14 +16,21 @@ class CallbackTest extends BitcoinSAsyncTest {
     val promise = Promise[Assertion]()
 
     val f1: Callback[Unit] = _ => {
-      Thread.sleep(testTimeout.toMillis)
-      promise.complete(fail("2nd callback did not start before timeout"))
+      system.scheduler.scheduleOnce(testTimeout) {
+        if (!promise.isCompleted) {
+          promise.complete(fail("2nd callback did not start before timeout"))
+        }
+      }
       FutureUtil.unit
```

Comment:
Now this isn't actually blocking anything, you need to return 

`promise.future`